packages/core/state/metadata/logics.ts

```diff
@@ -108,6 +114,60 @@ const receiveAnnotationsLogic = createLogic({
     type: RECEIVE_ANNOTATIONS,
 });
 
+const createNewAnnotationLogic = createLogic({
+    async process(deps: ReduxLogicDeps, dispatch, done) {
+        const { getState, httpClient, action } = deps;
+        const { annotation, annotationOptions } = action.payload;
+        const annotationProcessId = annotation.name;
+        dispatch(
+            interaction.actions.processStart(
+                annotationProcessId,
+                `Creating annotation ${annotation.name}...`
+            )
+        );
+
+        const annotationService = interaction.selectors.getAnnotationService(getState());
+        const applicationVersion = interaction.selectors.getApplicationVersion(getState());
+        if (annotationService instanceof HttpAnnotationService) {
+            if (applicationVersion) {
+                annotationService.setApplicationVersion(applicationVersion);
+            }
+            annotationService.setHttpClient(httpClient);
+        }
+
+        return new Promise<AnnotationResponseMms[]>(async (resolve, reject) => {
```

Comment:
Hmmm do we want to `return` here or should we be `awaiting` the `Promise`? I'm uncertain if it actually matters...?

Reply:
changed to `await` in [b3264d1](https://github.com/AllenInstitute/biofile-finder/pull/406/commits/b3264d1b8ea39f7b15e1cf385b839c9561ee0688). I think for our dispatches it doesn't matter, but `await` is more consistent with what we do elsewhere